content/blog/2016/2016-05-31-Jenkins-Usage-Statistics-Analysis-Intro.adoc

```diff
@@ -0,0 +1,98 @@
+technologies for example Knowledge Graphs, BigTable, SPARQL, RDF in Google. I am a passionate computer science student who
+is always interested in learning and looking for new challenges and technologies.That’s how I came across to Google Summer 
+of Code where I am working on some exciting data mining problems which you are going to encounter below in this blog.
+
+=== Project Overview
+Jenkins has collected anonymous jenkins usage information of more than 100,000 installations which includes set of plugins
+and their versions, release history etc. This data collection can be exploited for various data mining experiments. 
+Main goal of this project is to perform various analysis and studies over the available dataset to discover trend in data
+usage for example plugins utilization, spotting downgrades etc. This project covers a wide range of high priority problems
+which need to be solved such as:
+* _How quickly users are upgrading jenkins and its plugins_, will give us the insight of popularity of the releases.
+* _Spotting downgrades_, which will warn us that something is wrong with the version.
+* _Correlating_ what users are saying (community rating) with what users are doing (upgrades/downgrades).
+* _Distribution of cluster size_, where clusters represents jobs, nodes count which approximates the size of installation.
+* _Finding set of plugins which are likely to be used together_, will setup pillar for plugin recommendation system.
+
+As a part of the link:https://summerofcode.withgoogle.com/[Google Summer of Code] 2016, I will be working on the above
+mentioned problems. My mentors for the project are Kohsuke Kawaguchi and Daniel Beck. Few link:http://stats.jenkins-ci.org/jenkins-stats/svg/svgs.html[analyses] has been done over this
+data but those are outdated as charts can be more clearer and interactive. This project aims to improvise existing
+statistics and generating new ones discussed above.
+
+=== Use Cases
+This project covers wide-range of the use-cases that has been derived from the problems mentioned above.
+
+==== UseCase 1: Upgrade/Downgrade Analysis
+Understanding the trend in upgrades and downgrades have lots of utilities, some of them have already been explained
+earlier which includes measuring the popularity, spotting downgrades, giving warning about the wrong versions quickly etc.
+
+===== Usecase 1.1: Upgrading time metrics
+This analysis refers to the time taken by the users to upgrade its jenkins installation and its plugins. Required dataset
+for the analysis are link:https://bigquery.cloud.google.com/table/jenkins-user-stats:jenkinsstats.jenkins_usage?_ga=1.195703460.947004610.1452882463&pli=1[jenkins-usage] data and link:http://mirror.xmission.com/jenkins/updates/current/release-history.json[release-history] data. Types of queries that are supported in this
+analysis are:
+* *Query1 :* Enter the update release date, whose performance user wants to study.
```

Comment:
Not clear how the date is relevant here. If I want to know about a specific plugin, the system should show all releases. If I want to know about a specific version, the system should know when it was released.